### DISCUSSION TITLE
Is there a shortcut for open a URL in the current cell?

### DISCUSSION THREAD

**Discussion by reagle** [saulpw/visidata#1543] [Q&A] [answered]
Is there a shortcut for open a URL in the current cell? (Wish there was a command palette!)

**Comment by geekscrapy** [comment, marked as answer]
For the command palette, try z ctrl+H

For opening a URL, you might want `zo` to open the URL with vd, or if you want to open with the OS command try the below in your `.visidatarc`

```python
@VisiData.global_api
def sys_open_resource(vd, resource):
  import subprocess, os, platform
  if platform.system() == 'Darwin':       # macOS
      subprocess.call(('open', resource))
  elif platform.system() == 'Windows':    # Windows
      os.startfile(resource)
  else:                                   # linux variants
      subprocess.call(('xdg-open', resource))

# Generic OS open
Sheet.addCommand(None, 'sysopen-cell', 'sys_open_resource(cursorValue)', 'Open cell with OS')
```

Hope that helps!

**Reply by reagle** [reply]
Thanks @geekscrapy! Yes, I'm familiar with `z ctrl+H; g/` but that's a lot of keystrokes and not nearly as nice as a [genuine command palette with dynamic fuzzy match](https://github.com/saulpw/visidata/issues/247).

**Reply by saulpw** [reply]
I am hearing your request for a command palette, @reagle.  It's definitely on the list, but is a huge amount of work to get right, so it'll have to wait for when I have a stretch of time to devote to it.

**Reply by frosencrantz** [reply]
`Plugin 💡`: I thought it would be possible to create a plugin that uses `fzf` maybe with a `tmux` pane or popup and possibly `rich-cli` for formatting  to show the contents of the Command Sheet.   Use the strength of `fzf` to quickly and fuzzily select a command from the output..  That might be something that could be done quickly until Saul has time to create a builtin solution.
